.github/workflows/CI.yml

```diff
@@ -56,4 +51,22 @@ jobs:
         if: matrix.version == '1' && matrix.os == 'ubuntu-latest' && matrix.num_threads == 1
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
-          path-to-lcov: lcov.info
\ No newline at end of file
+          path-to-lcov: lcov.info
+
+  docs:
+    name: Documentation
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: '1'
+      - run: |
+          julia --project=docs -e '
+            using Pkg
+            Pkg.develop(PackageSpec(path=pwd()))
+            Pkg.instantiate()'
+      - run: julia --project=docs docs/make.jl
+        env:
```

Comment:
Awesome (and huge) list of comments! This is my favorite since it even fixes the problems I had on my local (NixOS) machine. I now have to work on a paper and will work on this PR again later today or tomorrow.